python_modules/libraries/dagster-shared/dagster_shared/cli/__init__.py

```diff
@@ -0,0 +1,198 @@
+                + ("each" if allow_multiple else "the")
+                + " file as a code location in the current python environment."
+            ),
+            envvar="DAGSTER_PYTHON_FILE",
+        ),
+        click.option(
+            "--module-name",
+            "-m",
+            multiple=allow_multiple,
+            help=(
+                "Specify module "
+                + ("or modules (flag can be used multiple times) " if allow_multiple else "")
+                + "where dagster definitions reside as top-level symbols/variables and load "
+                + ("each" if allow_multiple else "the")
+                + " module as a code location in the current python environment."
+            ),
+            envvar="DAGSTER_MODULE_NAME",
+        ),
+        click.option(
+            "--package-name",
+            multiple=allow_multiple,
+            help="Specify Python package where repository or job function lives",
+            envvar="DAGSTER_PACKAGE_NAME",
+            hidden=hide_legacy_options,
+        ),
+        click.option(
+            "--attribute",
+            "-a",
+            help=(
+                "Attribute that is either a 1) repository or job or "
+                "2) a function that returns a repository or job"
+            ),
+            envvar="DAGSTER_ATTRIBUTE",
+            hidden=hide_legacy_options,
+        ),
+    ]
+
+
+def generate_grpc_server_options() -> Sequence[ClickOption]:
+    return []
+
+
+def generate_workspace_options(*, hide_uncommon_options: bool) -> Sequence[ClickOption]:
+    return [
+        click.option("--empty-workspace", is_flag=True, help="Allow an empty workspace"),
+        click.option(
+            "--workspace",
+            "-w",
+            multiple=True,
+            type=click.Path(exists=True),
+            help="Path to workspace file. Argument can be provided multiple times.",
+        ),
+        *generate_python_pointer_options(
+            allow_multiple=True,
+            hide_legacy_options=hide_uncommon_options,
+        ),
+        click.option(
+            "--grpc-port",
+            type=click.INT,
+            required=False,
+            help="Port to use to connect to gRPC server",
+            hidden=hide_uncommon_options,
+        ),
+        click.option(
+            "--grpc-socket",
+            type=click.Path(),
+            required=False,
+            help="Named socket to use to connect to gRPC server",
+            hidden=hide_uncommon_options,
+        ),
+        click.option(
+            "--grpc-host",
+            type=click.STRING,
+            required=False,
+            help="Host to use to connect to gRPC server, defaults to localhost",
+            hidden=hide_uncommon_options,
+        ),
+        click.option(
+            "--use-ssl",
+            is_flag=True,
+            default=False,
+            help="Use a secure channel when connecting to the gRPC server",
+            hidden=hide_uncommon_options,
+        ),
```

Comment:
i hid these grpc options too under the premise that while useful in rare specific situations its preferable to not clutter `dg dev --help`